### ISSUE TITLE
Expose Firebase auth as a stream like the GoogleSignIn

### ISSUE THREAD

**Issue by branflake2267** [flutter/flutter#10836]
Could you expose the Firebase auth as a stream like the GoogleSign plugin does. 

https://stackoverflow.com/questions/44618334/how-can-i-test-if-the-user-is-signed-in-and-if-not-route-to-another-page-in-my-f/44620570?noredirect=1#comment76254962_44620570

```
Widget build() {
  return new StreamBuilder(
    stream: googleSignIn.onCurrentUserChanged,
    builder: (context, user) {
      return user.data != null ? new HomePage() : new SignInPage();
    }
  );
}
```

**Comment by collinjackson** [comment]
This was fixed in flutter/plugins#212

**Comment by kroikie** [comment]
@branflake2267

This issue has been moved to https://github.com/FirebaseExtended/flutterfire/issues/991. Any further collaboration will be done there.
